kernel/cooking.mli

```diff
@@ -27,13 +27,20 @@ type 'opaque result = {
   cook_flags : typing_flags;
 }
 
-val cook_constant : recipe -> cooking_info Opaqueproof.opaque result
-val cook_constr : cooking_info list ->
-  (constr * unit Opaqueproof.delayed_universes) -> (constr * unit Opaqueproof.delayed_universes)
+val cook_opaque_proofterm : cooking_info list ->
+  Opaqueproof.opaque_proofterm -> Opaqueproof.opaque_proofterm
+
+val cook_constant :
+  recipe -> cooking_info Opaqueproof.opaque result
 
 val cook_inductive :
   cooking_info -> mutual_inductive_body -> mutual_inductive_body
 
-(** {6 Utility functions used in module [Discharge]. } *)
+val make_cooking_info : expand_info -> named_context -> Univ.UContext.t -> cooking_info
+  (** Abstract a context assumed to be de-Bruijn free for terms and universes *)
+
+val lift_poly_univs : cooking_info -> Univ.AbstractContext.t -> cooking_info * Univ.AbstractContext.t
+
+val abstract_rel_context : cooking_info -> rel_context -> rel_context
 
-val expmod_constr : work_list -> constr -> constr
+val discharge_proj_repr : abstr_inst_info -> Names.Projection.Repr.t -> Names.Projection.Repr.t
```

Comment:
The terminology, `cook`, or `discharge`, or `abstract`, or `lift` should probably be simplified.
- `lift_poly_univs` and `abstract_rel_context` do morally the same: they traverse an already-quantified context, respectively a context of universe and a context of terms, updating the cooking info for the remainder of the judgement, and "cooking" this context at the same time.